src/groupByKey.ts

```diff
@@ -2,6 +2,12 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: object[], key: string): GroupsMap<object> {
+  const groupedItems: GroupsMap<object> = {};
```

Comment:
why `object`?
![image](https://user-images.githubusercontent.com/64437769/148390774-4f25b2f2-a548-43b7-bf29-a23d95b02996.png)

Reply:
But the key actually is a string "groupByKey(items: object[], **key: string**)" The object in "GroupsMap<object>" goes to the value type instead of T: 

type GroupsMap<T> = {
  [key: string]: T[];
};

And this value should be an array of objects, according to an example:

// Group books by color
groupByKey(books, 'color') === {
  'red': [
    { id: 1, color: 'red', country: 'Ukraine' },
    { id: 1, color: 'red', country: 'Italy' },
  ],
  'green': [
    { id: 1, color: 'green', country: 'Ukraine' },
  ],
}